### ISSUE TITLE
search

### ISSUE THREAD

**Issue by sungwoncho** [dnote-io/cli#93]
Should be able to search notes by content (full text search?)

```bash
$ dnote search "some content"
$ dnote search -b js "some content"
```

**Comment by michielboekhoff** [comment]
Is this going to be a fuzzy search? Also, just out of curiosity, what does the "-b" flag do?

**Comment by sungwoncho** [comment]
It would be cool to have a fuzzy search. Not sure how useful it can be though. I'd like to release a simple full text search and use it for a while to determine if we need fuzzy search.

I am thinking `-b` can search within a specified book.

**Comment by sungwoncho** [comment]
This is in progress in [find](https://github.com/dnote/cli/tree/find) branch. Incidentally, I have decided to name the command `find` because `s` is already a short for `sync`, and the search command needs a reasonable shortcut. `find` and `f` seemed reasonable.

e.g.

```
dnote f rpoplpush
dnote f "building a heap"
dnote find "merge sort" -b algorithm
```

feedback welcomed

**Comment by sungwoncho** [comment]
Finding book by its label might not be needed because number of books << number of notes. Also I think users can just `grep` for book name if need be without complicating API further. For what it's worth, though, here is an idea for an interface for searching books:

```
dnote f java --target book
# maybe `target` flag can default to `note`
```

**Comment by alxjsn** [comment]
Just wanted to mention that I like the fuzzy search idea brought up earlier in this thread. There are some notes that I have that I don't find right away because it might be a compound word or similar. For example the term `ghostscript` doesn't get captured by a search for `ghost` or `script`. Or for the word `outputfile`, I don't get any hits by searching for `output` or `file`. Those are both real terms I have in my notes that I wasn't able to find quickly.

**Comment by sungwoncho** [comment]
@alxjsn good idea. I encountered similar problems. Let's track that feature in #415.
